modules/ROOT/pages/writing.adoc

```diff
@@ -672,7 +678,229 @@ CREATE CONSTRAINT FOR (p:Person) REQUIRE (p.surname) IS UNIQUE
 
 *Take into consideration that the first label is used for the index creation.*
 
-=== Script option
+
+==== Approach
+
+The connector allows to enforce the following constraints:
+
+.Allowed configuration
+|===
+|config |values |default |description
+
+|`schema.optimization.node.unique` |true/false |false | Create the `UNIQUE` constraint for the properties defined in the option `node.keys`
+|`schema.optimization.node.key` |true/false |false | Create the `NODE KEY` constraint for the properties defined in the option `node.keys`
+|`schema.optimization.relationship.unique` |true/false |false | Create the `UNIQUE` constraint for the properties defined in the option `relationship.keys`
+|`schema.optimization.relationship.key` |true/false |false | Create the `RELATIONSHIP KEY` constraint for the properties defined in the option `relationship.keys`
+|`schema.type.constraint` |true/false |false | Creates the type constraints for node/relationships enforcing the type from the DataFrame schema
+|===
+
+===== Node constraints
+
+====== Using `schema.optimization.node.unique`
+
+Given the following example:
+
+[source, scala]
+----
+    ds.write
+      .format(classOf[DataSource].getName)
+      .mode(SaveMode.Overwrite)
+      .option("url", SparkConnectorScalaSuiteIT.server.getBoltUrl)
+      .option("labels", ":Person:Customer")
+      .option("node.keys", "surname")
+      .option("schema.optimization.node.unique", true)
+      .save()
+----
+
+Under the hood the Spark connector will create the following constraint:
+
+`CREATE CONSTRAINT `spark_NODE_UNIQUE-CONSTRAINT_Person_surname` IF NOT EXISTS FOR (e:Person) REQUIRE (e.surname) IS UNIQUE`
```

Comment:
might be worth calling out that the first label is selected (it is explicitly mentioned in the deprecated schema section)

Reply:
applies to all sections below

Reply:
Done